pkg/karmadactl/cmdinit/bootstraptoken/clusterinfo/clusterinfo_test.go

```diff
@@ -0,0 +1,208 @@
+/*
+Copyright 2024 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterinfo
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"testing"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/client-go/kubernetes"
+	clientset "k8s.io/client-go/kubernetes"
+	fakeclientset "k8s.io/client-go/kubernetes/fake"
+	bootstrapapi "k8s.io/cluster-bootstrap/token/api"
+
+	"github.com/karmada-io/karmada/operator/pkg/certs"
+)
+
+func TestCreateBootstrapConfigMapIfNotExists(t *testing.T) {
+	kubeAdminConfig := `
+apiVersion: v1
+clusters:
+- cluster:
+    certificate-authority-data: %s
+    server: https://test-cluster:6443
+  name: test-cluster
+contexts:
+- context:
+    cluster: test-cluster
+    user: test-user
+  name: test-context
+current-context: test-context
+kind: Config
+preferences: {}
+users:
+- name: test-user
+  user:
+    client-certificate-data: %s
+    client-key-data: %s
+`
+	tests := []struct {
+		name    string
+		client  clientset.Interface
+		cfgFile string
+		prep    func(cfgFile string) error
+		verify  func(clientset.Interface) error
+		cleanup func(cfgFile string) error
+		wantErr bool
+		errMsg  string
+	}{
+		{
+			name:    "CreateBootstrapConfigMapIfNotExists_NonExistentConfigFile_FailedToLoadAdminKubeConfig",
+			prep:    func(string) error { return nil },
+			verify:  func(clientset.Interface) error { return nil },
+			cleanup: func(string) error { return nil },
+			wantErr: true,
+			errMsg:  "failed to load admin kubeconfig",
+		},
+		{
+			name:    "CreateBootstrapConfigMapIfNotExists_WithConfigFile_ConfigMapCreatedInKubePublicNamespace",
+			client:  fakeclientset.NewSimpleClientset(),
```

Comment:
```suggestion
			client: fakeclientset.NewClientset(),
```
`NewSimpleClientset` is deprecated, and can be replaced by `NewClientset`.

Reply:
It seems `NewClientset` is not accessible on `fakeclientset`?

Reply:
I'm sorry, I don't get it. Can you elaborate?

Reply:
Never mind. It was a vendor cache issue on my side, where an older version of the fake Kubernetes library was causing the problem. Now, I can access `NewClientset` 👍.

Reply:
I think this could be a good opportunity to update it across the entire codebase. I might submit a follow-up PR for this. What do you think?

Reply:
I think this could be a good opportunity to update it across the entire codebase. I might submit a follow-up PR for this. 👍

Reply:
good idea~